### DISCUSSION TITLE
The imported file was executed twice。

### DISCUSSION THREAD

**Discussion by mrhanCH** [zauberzeug/nicegui#2300] [Q&A] [answered]
### Description

My code:
```py
from nicegui import run, ui
import testplay

def test():
    print(1)

ui.button('play', on_click=test)
ui.run(port=8087)
```

The import file -> testplay.py:
```py
print(0)
```
Output results:
```
0
0
NiceGUI ready to go on http://localhost:8087, and http://192.168.0.112:8087
```

So, why was the imported content executed twice?
Thanks!

**Comment by falkoschindler** [comment, marked as answer]
Hi @mrhanCH!

I think this FAQ answers your question:
https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-code-executed-twice

> You are probably using `reload=True` which runs the main code once and then spawns a subprocess which is killed and relaunched on file change. To avoid evaluation of your code in the first "init" you have several options:
[...]
